cairo/tests/ethereum/cancun/vm/instructions/test_environment.py

```diff
@@ -0,0 +1,169 @@
+    def test_origin(self, cairo_run_py, evm: Evm):
+        try:
+            cairo_result = cairo_run_py("origin", evm)
+        except Exception as cairo_error:
+            with strict_raises(type(cairo_error)):
+                origin(evm)
+            return
+
+        origin(evm)
+        assert evm == cairo_result
+
+    @given(evm=EvmBuilder().with_gas_left().with_env(environment_empty_state).build())
+    def test_caller(self, cairo_run, evm: Evm):
+        try:
+            cairo_result = cairo_run("caller", evm)
+        except Exception as cairo_error:
+            with strict_raises(type(cairo_error)):
+                caller(evm)
+            return
+
+        caller(evm)
+        assert evm == cairo_result
+
+    @given(evm=EvmBuilder().with_gas_left().with_env(environment_empty_state).build())
+    def test_callvalue(self, cairo_run, evm: Evm):
+        try:
+            cairo_result = cairo_run("callvalue", evm)
+        except Exception as cairo_error:
+            with strict_raises(type(cairo_error)):
+                callvalue(evm)
+            return
+
+        callvalue(evm)
+        assert evm == cairo_result
+
+    @given(evm=EvmBuilder().with_gas_left().with_env(environment_empty_state).build())
+    def test_codesize(self, cairo_run, evm: Evm):
+        try:
+            cairo_result = cairo_run("codesize", evm)
+        except Exception as cairo_error:
+            with strict_raises(type(cairo_error)):
+                codesize(evm)
+            return
+
+        codesize(evm)
+        assert evm == cairo_result
+
+    @given(evm=EvmBuilder().with_gas_left().with_env(environment_empty_state).build())
+    def test_gasprice(self, cairo_run, evm: Evm):
+        try:
+            cairo_result = cairo_run("gasprice", evm)
+        except Exception as cairo_error:
+            with strict_raises(type(cairo_error)):
+                gasprice(evm)
+            return
+
+        gasprice(evm)
+        assert evm == cairo_result
+
+    @given(evm=EvmBuilder().with_gas_left().with_env(environment_empty_state).build())
+    def test_returndatasize(self, cairo_run, evm: Evm):
+        try:
+            cairo_result = cairo_run("returndatasize", evm)
+        except Exception as cairo_error:
+            with strict_raises(type(cairo_error)):
+                returndatasize(evm)
+            return
+
+        returndatasize(evm)
+        assert evm == cairo_result
+
+    @given(evm=EvmBuilder().with_gas_left().with_env(environment_empty_state).build())
+    def test_self_balance(self, cairo_run, evm: Evm):
+        try:
+            cairo_result = cairo_run("self_balance", evm)
+        except Exception as cairo_error:
+            with strict_raises(type(cairo_error)):
+                self_balance(evm)
+            return
+
+        self_balance(evm)
+        assert evm == cairo_result
+
+    @given(evm=EvmBuilder().with_gas_left().with_env(environment_empty_state).build())
```

Comment:
The same strategy is used for all the test 
Could be put at the beginning of the file